File: legacy/python-backend/app/repositories/order_repository.py
from sqlalchemy import select, desc
from sqlalchemy.ext.asyncio import AsyncSession

from app.models.order import Order


class OrderRepository:

    def __init__(
        self,
        db: AsyncSession
    ):
        self.db = db

    async def create(
        self,
        order: Order
    ) -> Order:

        self.db.add(order)

        await self.db.commit()

        await self.db.refresh(order)

        return order

    async def get_by_id(
        self,
        order_id: str
    ):

        result = await self.db.execute(
            select(Order).where(
                Order.id == order_id
            )
        )

        return result.scalar_one_or_none()

    async def get_recent(
        self,
        limit: int = 20
    ):
        result = await self.db.execute(
            select(Order)
            .order_by(desc(Order.created_at))
            .limit(limit)
        )
        return result.scalars().all()

File: backend/src/repositories/orderRepository.ts
import { prisma } from "../core/database";
import { Order } from "@prisma/client";

export class OrderRepository {
  static async create(data: {
    id: string;
    symbol: string;
    side: string;
    order_type: string;
    quantity: number;
    price: number;
    status: string;
    exchange_order_id?: string | null;
  }): Promise<Order> {
    return prisma.order.create({
      data,
    });
  }

  static async getById(orderId: string): Promise<Order | null> {
    return prisma.order.findUnique({
      where: {
        id: orderId,
      },
    });
  }

  static async getRecent(limit: number = 20): Promise<Order[]> {
    return prisma.order.findMany({
      orderBy: {
        created_at: "desc",
      },
      take: limit,
    });
  }
}
export default OrderRepository;
